pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentRestletResource.java

```diff
@@ -230,19 +232,38 @@ protected Representation toggleSegmentState(String tableName, String segmentName
 
     JSONArray ret = new JSONArray();
     String tableNameWithType = null;
-    if ((tableType == null || TableType.OFFLINE.name().equalsIgnoreCase(tableType))
+    List<String> segmentsToToggle = new ArrayList<>();
```

Comment:
maybe it is best to introduce a constraint here that tabletype cannot be null if both types are present. This way, we can limit damages of accidentally toggling states of all segments,  when this api is used for 'drop'.